src/main/java/org/simplejavamail/outlookmessageparser/model/OutlookMessage.java

```diff
@@ -313,9 +313,7 @@ private void checkToRecipient() {
 			for (final OutlookAttachment attachment : getOutlookAttachments()) {
 				if (attachment instanceof OutlookFileAttachment) {
 					final OutlookFileAttachment fileAttachment = (OutlookFileAttachment) attachment;
-					if (!tryAddCid(cidMap, html, fileAttachment, fileAttachment.getFilename())) {
-						tryAddCid(cidMap, html, fileAttachment, fileAttachment.getLongFilename());
-					}
+					tryAddCid(cidMap, html, fileAttachment, fileAttachment.getContentId());
```

Comment:
You removed the failover for using alternative cid's in favor of Outlook native 3712 attribute. Is that guaranteed to be always present?

Reply:
To be honest I don't know, I just assumed it does. We can also keep the old failovers too, just to be sure.

Reply:
Yeah it's better to keep the fallback just in case. If you could please amend it I'll merge and release.